src/modules/peek/Peek.FilePreviewer/FilePreview.xaml.cs

```diff
@@ -149,13 +149,6 @@ private async Task OnItemPropertyChanged()
                 BrowserPreview.Visibility = Visibility.Collapsed;
                 ArchivePreview.Visibility = Visibility.Collapsed;
                 UnsupportedFilePreview.Visibility = Visibility.Collapsed;
-
-                ImagePreview.FlowDirection = FlowDirection.LeftToRight;
-                VideoPreview.FlowDirection = FlowDirection.LeftToRight;
-                BrowserPreview.FlowDirection = FlowDirection.LeftToRight;
-                ArchivePreview.FlowDirection = FlowDirection.LeftToRight;
-                UnsupportedFilePreview.FlowDirection = FlowDirection.LeftToRight;
-
```

Comment:
Why did you only move the VideoPreview and BrowserControl flowdirection values ​​to the xaml file and leave the others as they are?

Some modifications were made for the titlebar for RTL languages. I saw that TitleBar files are removed and those changes were not there in the new changes in MainWindow files. If there is no generic solution for this in new changes, old bugs may pop up again.

I followed the conversation, there are commits in some parts, if you can explain exactly where you need help, I can do a more detailed analysis and try to help.

Reply:
The thing that renders windows and controls (is it XAML or SDK or WinUI or..?) deals with the right-to-left layout. As it should. You can try by building a window with a few controls, optionally in a grid, and then set your Windows language to Hebrew. The manual work you introduced in the past is unnecessary and cumbersome.

The only thing that actually needs adjustment is the _content_ of the controls that you comment on now: text and videos can be _mirrored_ otherwise.

Reply:
Also, I merged the Title Bar code with the Main window. Now the problem is that the text controls in xaml don't print/update according to the values they should display. Sam said to use a dependency for that, but I don't have the knowledge to place the exact code in the right place.